### ISSUE TITLE
Deletion of a used reference causes server error

### ISSUE THREAD

**Issue by thclark** [thclark/wagtail_references#9]
If I delete a reference which has a protected foreign key relationship to another model, I get a server error 500 page.

The model (in the users app) looks something like:
```python
class SomePage(Page):
    reference = models.OneToOneField(get_reference_model_string(), on_delete=models.PROTECT)
```

The logs are as follows:
```
2019-05-17T19:25:58.837019+00:00 app[web.1]: django.db.models.deletion.ProtectedError: ("Cannot delete some instances of model 'Reference' because they are referenced through a protected foreign key: 'LegislationPage.citation'", <QuerySet [<LegislationPage: The Woodstock Agreement>]>)
2019-05-17T19:25:58.838803+00:00 app[web.1]: 10.12.254.40 - - [17/May/2019:19:25:58 +0000] "POST /admin/cms/references/1/delete/ HTTP/1.1" 500 27 "https://tpast.herokuapp.com/admin/cms/references/1/delete/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.157 Safari/537.36"
2019-05-17T19:25:58.835501+00:00 heroku[router]: at=info method=POST path="/admin/cms/references/1/delete/" host=tpast.herokuapp.com request_id=05d4fa48-2235-4765-a324-468b0b5472e7 fwd="82.8.129.188" dyno=web.1 connect=1ms service=537ms status=500 bytes=512 protocol=https
2019-05-17T19:25:59.418561+00:00 app[web.1]: 10.12.254.40 - - [17/May/2019:19:25:59 +0000] "GET /favicon.ico HTTP/1.1" 200 12467 "https://tpast.herokuapp.com/admin/cms/references/1/delete/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.157 Safari/537.36"
```

The desired behaviour is to display a friendly message to the user that the reference is in use!

Related to an open wagtail issue  wagtail/wagtail#1602

**Comment by thclark** [comment]
Waiting for wagtail/wagtail#4702 to be pulled in and released. This will be used to show to the user, in the confirm-deletion view, what other usages there are and therefore what other items will be affected.

That way, you can be comfortable using ``SET_NULL`` instead of ``PROTECT`` knowing that at the user can see what the result of their deletion will be.

**Comment by lb-** [comment]
Update - this PR has resolved part of this issue, relating to the server error handling
https://github.com/wagtail/wagtail/pull/5437

**Comment by thclark** [comment]
Brilliant, thanks for the update @lb. Closing this issue since the referenced PR was released in wagtail 2.6 and the remaining part of the issue is wagtail-only.

Fix is simple: Update to wagtail >= 2.6
